pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentLongPairSet.java

```diff
@@ -331,7 +331,11 @@ private static final class Section extends StampedLock {
         boolean contains(long item1, long item2, int hash) {
             long stamp = tryOptimisticRead();
             boolean acquiredLock = false;
-            int bucket = signSafeMod(hash, capacity);
+
+            // add local variable here, so OutOfBound won't happen
+            long[] table = this.table;
+            // calculate table.length / 2 as capacity to avoid rehash changing capacity
+            int bucket = signSafeMod(hash, table.length / 2);
```

Comment:
similar comment here about requesting the rationale for using 2 as the divisior.